forwardproxy.go

```diff
@@ -284,7 +284,7 @@ func (fp *ForwardProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int,
 			panic("There was a check for http version, yet it's incorrect")
 		}
 	} else {
-		outReq, err := fp.generateForwardRequest(r)
+		outReq, err := fp.decorateOriginalRequest(r)
```

Comment:
Since this is explicitly a mutator, I think we should probably get rid of `outReq` entirely.